dev/MAINTAINING.md

```diff
@@ -26,71 +26,98 @@
 * If this is a major release check if any previously deprecated features are to
   be removed in this release.
 * Check pre-release APIs to be exposed, especially from streamly-examples
-* PVP is ahered to, do not change pre-release APIs in a minor release
+* PVP is adhered to, additionally do not change pre-release APIs in a
+  minor release
 
 * _Documentation_:
 
-    * README is updated
+    * README is updated, links in README are stable. Run `rg '/blob/'`
+      and `rg '/tree/'` to search for github links, replace `master` ref
+      with release tag where possible (e.g. streamly-examples). Ideally,
+      the distribution process should automatically replace these with
+      current stable references.
     * Haddock docs are consistent with the changes in the release
     * Tutorial has been updated for new changes
     * Documents in the `docs` directory are consistent with new changes
     * All combinators have time and space complexity annotations
     * All combinators have `since` annotation
+    * Modules that were renamed or APIs that moved to other modules should be
+      marked as "since" the current release.
+    * Check all released modules to ensure that they do not have any leftover
+      `Internal` or `Pre-release` annotations.
     * Unreleased combinators should be marked with `Pre-release` or
       `Internal` annotations
+    * Streamly homepage is updated with the release docs
+      * FileSystem.Event.Windows/Darwin modules require special handling
 
 * _Build and Test_:
 
     * All CIs are green
+    * Run tests using bin/test.sh to test with memory restrictions
     * Manually build and test for all flags (esp. `dev` flag) not covered by CIs
-    * Test latest GHC version with -O0 and -O1
+    * Test prime GHC version with -O0 and -O1
 
 * _Benchmarks_:
 
     * Check regressions from previous release
-    * Run benchmarks with large stream size (`bench.sh -- long`) to
+    * Run benchmarks with large stream size (`bench.sh --long`) to
       check for space leaks and to ensure constant memory usage for streaming
       operations.
     * Run benchmarks with `dev` flag on. Some `fileio` benchmarks are disabled
       in regular runs.
     * Check comparative benchmarks using streaming-benchmarks
 
-* _Examples_:
+* _Dependencies_:
+
+    * Make sure all dependency bounds can use latest versions
+    * Update `stack.yaml` to latest stable resolver, cleanup extra-deps
+    * Make sure fusion-plugin is up to date and uploaded
+    * Check the dependency footprint of the library. See
+      [Generating Dependency Graphs](##generating-dependency-graphs) section.
+
+* _Dependent Packages_:
 
     * Update
       [streamly-examples](https://github.com/composewell/streamly-examples)
       to make sure it runs with the latest release.
     * Check the performance of examples where applicable
+    * Update streamly-bytestring
+    * Consider updating other packages e.g. streamly-process, streamly-lz4
```

Comment:
We **should** update other packages. We should track all the packages that are required to be updated. Atleast the important ones. For now, `streamly-bytestring` is the only important one that has been released.
`unicode-transforms` is also, I believe, pretty important.

Reply:
These are only the packages that depend on streamly. unicode-transforms does not depend on streamly.